src/statistics/slice_statistics.rs

```diff
@@ -61,48 +60,38 @@ impl OrderStatistics<f64> for [f64] {
     fn ranks(&mut self, tie_breaker: RankTieBreaker) -> Vec<f64> {
         let n = self.len();
         let mut ranks: Vec<f64> = vec![0.0; n];
-        let mut index: Vec<usize> = (0..n).collect();
-
+        let mut enumerated: Vec<_> = self.iter().enumerate().collect();
+        enumerated.sort_by(|fst, snd| fst.1.partial_cmp(&snd.1).unwrap());
```

Comment:
this is a bit incomprehensible, would it be cleaner to destructure the two tuples 

```sort_by(|(_, el_a), (_, el_b)| el_a.partial_cmp(el_b).unwrap())```